keel-ec2-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -169,14 +169,6 @@ data class ClusterSpec(
 
   override val artifactType: ArtifactType? = DEBIAN
 
-  // Returns the artifact name set by resolvers, or attempts to find the artifact name from the image provider.
-  override val artifactName: String?
-    get() = _artifactName
-      ?: when (imageProvider) {
-        is ArtifactImageProvider -> imageProvider.deliveryArtifact.name
-        else -> null
-      }
```

Comment:
Either set by the resolver from a reference, or not. No need for this complexity